crates/ruff_linter/src/rules/pylint/rules/useless_import_alias.rs

```diff
@@ -52,8 +65,76 @@ pub(crate) fn useless_import_alias(checker: &mut Checker, alias: &Alias) {
     if alias.name.as_str() != asname.as_str() {
         return;
     }
+    // A required import with a useless alias causes an infinite loop.
+    // See https://github.com/astral-sh/ruff/issues/14283
+    if checker
+        .settings
+        .isort
+        .requires_module_import(alias.name.to_string(), Some(asname.to_string()))
+    {
+        let diagnostic = Diagnostic::new(
+            UselessImportAlias {
+                required_import_conflict: true,
+            },
+            alias.range(),
+        );
+        checker.diagnostics.push(diagnostic);
+        return;
+    }
+
+    let mut diagnostic = Diagnostic::new(
+        UselessImportAlias {
+            required_import_conflict: false,
+        },
+        alias.range(),
+    );
+    diagnostic.set_fix(Fix::unsafe_edit(Edit::range_replacement(
+        asname.to_string(),
+        alias.range(),
+    )));
+    checker.diagnostics.push(diagnostic);
+}
+
+/// PLC0414
+pub(crate) fn useless_import_from_alias(
+    checker: &mut Checker,
+    alias: &Alias,
+    module: Option<&str>,
+    level: u32,
+) {
+    let Some(asname) = &alias.asname else {
+        return;
+    };
+    if alias.name.contains('.') {
+        return;
+    }
+    if alias.name.as_str() != asname.as_str() {
+        return;
+    }
```

Comment:
I'm not sure I understand the suggestion because `required_member_import` might check for something like `import numpy` which is not aliased at all (and shouldn't cause an early return). This block seems to belong to the logic of "are you aliased and useless" rather than "are you required", right?

Reply:
Makes sense. I assumed that a similar check exists in the isort code and I wanted to deduplicate it. Thanks for explaining.